### PR TITLE
PT-1920 Return Special code on authorization errors in result

### PR DESCRIPTION
**Pull request by ksavosteev**
## Description
Returns 401 Unauthorized request status on wrong/expired/empty token instead of 200.

### CODE REVIEW

File: src/VirtoCommerce.ExperienceApiModule.Core/Infrastructure/Authorization/AuthorizationException.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+using System;
+
+namespace VirtoCommerce.ExperienceApiModule.Core.Infrastructure.Authorization
+{
+    public sealed class AuthorizationException : Exception
```
Comment (Vectorfield4) [inline src/VirtoCommerce.ExperienceApiModule.Core/Infrastructure/Authorization/AuthorizationException.cs:5]: Add XML-doc about this exception.

File: src/VirtoCommerce.ExperienceApiModule.Core/Middleware/AuthorizationErrorHandlingMiddleware.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Newtonsoft.Json;
+using VirtoCommerce.ExperienceApiModule.Core.Infrastructure.Authorization;
+
+namespace VirtoCommerce.ExperienceApiModule.Core.Middleware
+{
+    public class AuthorizationErrorHandlingMiddleware
```
Comment (Vectorfield4) [inline src/VirtoCommerce.ExperienceApiModule.Core/Middleware/AuthorizationErrorHandlingMiddleware.cs:9]: No interfaces?

File: src/VirtoCommerce.ExperienceApiModule.Core/Infrastructure/GraphQLExecuter.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        }
+
+        /// <summary>
+        /// Override to set option.ThrowOnUnhandledException to true
+        /// </summary>
+        protected override ExecutionOptions GetOptions(string operationName, string query, Inputs variables, IDictionary<string, object> context, IServiceProvider requestServices, CancellationToken cancellationToken)
+        {
+            var options = base.GetOptions(operationName, query, variables, context, requestServices, cancellationToken);
+
+            options.ThrowOnUnhandledException = true;
```
Comment (Vectorfield4) [inline src/VirtoCommerce.ExperienceApiModule.Core/Infrastructure/GraphQLExecuter.cs:34]: Add comment why we need this.

**Review by Vectorfield4 — APPROVED** [review]

File: src/VirtoCommerce.ExperienceApiModule.Core/Middleware/AuthorizationErrorHandlingMiddleware.cs (hunk 2 above)
Comment (ksavosteev) [reply src/VirtoCommerce.ExperienceApiModule.Core/Middleware/AuthorizationErrorHandlingMiddleware.cs:9]: Middlewares can be registered as convention-based, so no need for interfaces.

**Pull request merged**